api/src/main/java/com/velocitypowered/api/bossbar/BossBar.java

```diff
@@ -0,0 +1,145 @@
+package com.velocitypowered.api.bossbar;
+
+import com.velocitypowered.api.proxy.Player;
+import java.util.Collection;
+import java.util.UUID;
+import net.kyori.text.Component;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Represents a boss bar, which can be send to a (group of) player(s) <b>WARNING!!!</b> This works
+ * only on 1.9+. <b>YOU ARE ON YOUR OWN IF YOU USE THIS ON 1.8</b>
+ */
+public interface BossBar {
+
+  /**
+   * Adds all specified players to this boss bar
+   *
+   * @param players players
+   * @see #addPlayer(Player)
+   */
+  void addPlayers(@NonNull Iterable<Player> players);
+
+  /**
+   * Adds player to this boss bar. This adds the player to the {@link #getPlayers()} and makes him
+   * see the boss bar
+   *
+   * @param player the player you wish to add
+   */
+  void addPlayer(@NonNull Player player);
+
+  /**
+   * Removes player from this boss bar. This removes the player from {@link #getPlayers()} and makes
+   * him not see the boss bar
+   *
+   * @param player the player you wish to remove
+   */
+  void removePlayer(@NonNull Player player);
+
+  /**
+   * Removes all specified players from this boss bar
+   *
+   * @param players players
+   * @see #removePlayer(Player)
+   */
+  void removePlayers(@NonNull Iterable<Player> players);
+
+  /**
+   * Removes all players, that see this boss bar.
+   *
+   * @see #removePlayer(Player)
+   */
+  void removeAllPlayers();
+
+  /**
+   * Gets the title of this boss bar.
+   *
+   * @return title
+   */
+  @NonNull
+  Component getTitle();
+
+  /**
+   * Sets a new title of the boss bar.
+   *
+   * @param title new title
+   */
+  void setTitle(@NonNull Component title);
+
+  /**
+   * Gets the boss bar's progress. In Minecraft, this is called 'health' of the boss bar
+   *
+   * @return progress
+   */
+  float getProgress();
+
+  /**
+   * Sets a new progress of the boss bar. In Minecraft, this is called 'health' of the boss bar.
+   *
+   * @param progress a float between 0 and 1, representing boss bar's progress
+   * @throws IllegalArgumentException if the new progress is not between 0 and 1
+   */
+  void setProgress(float progress);
+
+  /**
+   * Returns a unmodifiable {@link Collection} of all {@link Player} added to the boss bar.
+   *
+   * @return players
+   */
+  @Nullable
+  Collection<Player> getPlayers();
+
+  /**
+   * Gets the {@link UUID} of the boss bar.
+   *
+   * @return uuid
+   */
+  @NonNull
+  UUID getUUID();
```

Comment:
Let's not expose this detail in the API, it's very protocol-specific. Who knows, Mojang may decide to change this and then this method has to be deprecated early.